course_content/notebooks/dask_intro.ipynb

```diff
@@ -393,6 +393,281 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "cubelist = iris.cube.CubeList(map(iris.load_cube, files))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "len(cubelist)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "type(cubelist)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Load multiple files into multiple cubes using `dask.bag`\n",
+    "\n",
+    "We can parallelise the loading of data from files into cubes by creating a `dask.bag` object which `map`s our list of filenames onto the function `iris.load_cube()`:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "bag_o_cubes = db.from_sequence(files).map(iris.load_cube)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "type(bag_o_cubes)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "bag_o_cubes.visualize()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "`bag_o_cubes` is a delayed/\"lazy\" object which we can `visualize` as a task graph and doesn't execute the computation of mapping filenames onto `iris.load_cube()` until we give it the command to do so (namely `bag_o_cubes.compute()`). You can see from the task graph (double-click to enlarge) that each file is being loaded into a cube invidually and in parallel.\n",
+    "\n",
+    "Since we want to generate a `CubeList` like the previous example we can compute `bag_o_cubes` from within our `iris.cube.CubeList()` command:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "bag_cubelist = iris.cube.CubeList(bag_o_cubes.compute())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "len(bag_cubelist)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "type(bag_cubelist)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "It is worth noting that part of the beauty of Python is that we do not necessarily need to create four objects (`fp`, `files`,  `bag_o_cubes`, `bag_cubelist`) to get a `CubeList` object but can chain the functions together into one big function call:\n",
+    "        \n",
+    "        iris.cube.CubeList(db.from_sequence(files).map(iris.load_cube).compute())"
```

Comment:
In this particular case it doesn't actually make any difference (as there's no special syntax to highlight), but in you can also make a preformatted code block as follows that includes syntax highlighting:
```
\```python
iris.cube.CubeList(db.from_sequence(files).map(iris.load_cube).compute())
\```
```

(But drop the leading slashes – I can't quite get the syntax highlighting working correctly, ironically enough.)

Reply:
#111 addresses :white_check_mark: